src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -38,8 +42,42 @@ class JooqBurdenEstimateRepository(
         stochasticBurdenEstimateWriter: StochasticBurdenEstimateWriter? = null
 ) : JooqRepository(dsl), BurdenEstimateRepository
 {
-    private val centralBurdenEstimateWriter: BurdenEstimateWriter = centralBurdenEstimateWriter ?:
-            CentralBurdenEstimateWriter(dsl)
+    private fun getCountriesAsLookup(): Map<Short, String> = dsl.select(Tables.COUNTRY.ID, Tables.COUNTRY.NID)
+            .from(Tables.COUNTRY)
+            .fetch()
+            .intoMap(Tables.COUNTRY.NID, Tables.COUNTRY.ID)
+
+    override fun validateEstimates(set: BurdenEstimateSet, expectedRowMap: HashMap<String, HashMap<Short, HashMap<Short, Boolean>>>): HashMap<String, HashMap<Short, HashMap<Short, Boolean>>>
+    {
+        val countries = getCountriesAsLookup()
+        dsl.select(BURDEN_ESTIMATE.COUNTRY, BURDEN_ESTIMATE.AGE, BURDEN_ESTIMATE.YEAR)
+                .from(BURDEN_ESTIMATE)
+                .where(BURDEN_ESTIMATE.BURDEN_ESTIMATE_SET.eq(set.id)).stream().use { stream ->
+                    stream.forEach { validate(it.into(BURDEN_ESTIMATE), countries, expectedRowMap) }
+                }
+
+        return expectedRowMap
+    }
+
+    private fun validate(r: BurdenEstimateRecord, countries: Map<Short, String>, expectedRows: HashMap<String, HashMap<Short, HashMap<Short, Boolean>>>)
+    {
+        val countryId = countries[r.country]
+        val ages = expectedRows[countryId]
+                ?: throw BadRequest("We are not expecting data for country $countryId")
```

Comment:
Technically none of these `BadRequest`s should ever be thrown because estimates have already been validated before populating the table. I left it in for like, added safety, but perhaps we can just assume no unexpected rows?